devicetypes/smartthings/aeotec-wallmote.src/aeotec-wallmote.groovy

```diff
@@ -0,0 +1,171 @@
+/**
+ *  Copyright 2019 SmartThings
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+
+import groovy.json.JsonOutput
+
+metadata {
+    definition (name: "Aeotec Wallmote", namespace: "smartthings", author: "SmartThings", ocfDeviceType: "x.com.st.d.remotecontroller", mcdSync: true) {
+        capability "Actuator"
+        capability "Button"
+        capability "Battery"
+        capability "Holdable Button"
+        capability "Configuration"
+        capability "Sensor"
+        capability "Health Check"
+
+        fingerprint mfr: "0086", model: "0082", deviceJoinName: "Aeotec Wallmote Quad", mnmn: "SmartThings", vid: "generic-4-button"
+        fingerprint mfr: "0086", model: "0081", deviceJoinName: "Aeotec Wallmote", mnmn: "SmartThings", vid: "generic-2-button"
+    }
+
+    tiles(scale: 2) {
+        multiAttributeTile(name: "rich-control", type: "generic", width: 6, height: 4, canChangeIcon: true) {
+            tileAttribute("device.button", key: "PRIMARY_CONTROL") {
+                attributeState "default", label: ' ', action: "", icon: "st.unknown.zwave.remote-controller", backgroundColor: "#ffffff"
+            }
+        }
+        valueTile("battery", "device.battery", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+            state "battery", label:'${currentValue}% battery', unit:""
+        }
+
+        main("rich-control")
+        details(["rich-control", childDeviceTiles("endpoints"), "battery"])
+    }
+}
+
+def getNumberOfButtons() {
+    def modelToButtons = ["0082" : 4, "0081": 2]
+    return modelToButtons[zwaveInfo.model] ?: 1
+}
+
+def installed() {
+    sendEvent(name: "numberOfButtons", value: numberOfButtons)
+    sendEvent(name: "button", value: "pushed", data: [buttonNumber: 1])
```

Comment:
Do we need to create an `supportedButtonValues` event too?

Reply:
I don't think so? @dkirker ?

Reply:
Yeah, with the addition of a zillion other button values in the enum there is now supportedButtonValues.